build.go

```diff
@@ -192,7 +192,11 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		}
 	}
 
-	return c.lifecycle.Execute(ctx, lifecycleOpts)
+	if err := c.lifecycle.Execute(ctx, lifecycleOpts); err != nil {
```

Comment:
I put the error here, rather than on `lifecycle.Analyze` or `lifecycle.Export`, because that would be easiest to test (by substituting a failing lifecycle).